src/heitech.zer0mqXt.core/Patterns/Socket.cs

```diff
@@ -118,20 +123,27 @@ public Socket(SocketConfiguration configuration)
                     catch (System.Exception ex)
                     {
                         // failure to parse or any other exception leads to a non successful response, which then in turn can be handled on the request side
-                        // todo exception propagation?
                         _configuration.Logger.Log(new ErrorLogMsg($"Responding to [Request:{typeof(T)}] with [Response:{typeof(TResult)}] did fail: " + ex.Message));
                         response = new RequestReplyMessage<TResult>(_configuration, default(TResult), success: false);
                     }
                     // try send response with timeout
                     bool noTimeout = rsSocket.TrySendMultipartMessage(_configuration.TimeOut, response);
                     if (!noTimeout)
                         _configuration.Logger.Log(new ErrorLogMsg($"Responding to [Request:{typeof(T)}] with [Response:{typeof(TResult)}] timed-out after {_configuration.TimeOut}"));
+
+                    if (respondOnce)
+                        runReponder = false;
                 }
-            });
+            }, UseBestToken(cancellationToken));
             // wait for the Set inside the background thread
             eventHandle.WaitOne();
         }
 
+        public void StopResponding() 
+        {
+
```

Comment:
should this set `this.runReponder` to `false`?

Reply:
no this method is no longer needed. I am going to remove it